x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals/actions.tsx

```diff
@@ -58,19 +74,117 @@ export const updateSignalStatusAction = async ({
   }
 };
 
-export const sendSignalsToTimelineAction = async ({
+export const sendSignalToTimelineAction = async ({
+  apolloClient,
   createTimeline,
-  data,
-}: SendSignalsToTimelineActionProps) => {
-  const stringFilter = data[0].filter(d => d.field === 'signal.rule.filters')?.[0]?.value ?? [];
+  ecsData,
+  updateTimelineIsLoading,
+}: SendSignalToTimelineActionProps) => {
+  const timelineId =
+    ecsData.signal?.rule?.timeline_id != null ? ecsData.signal?.rule?.timeline_id[0] : '';
 
-  // TODO: Switch to using from/to when adding dateRange
-  // const [stringFilters, from, to] = getFilterAndRuleBounds(data);
-  const parsedFilter = stringFilter.map(sf => JSON.parse(sf));
-  createTimeline({
-    id: 'timeline-1',
-    filters: parsedFilter,
-    dateRange: undefined, // TODO
-    kqlQuery: undefined, // TODO
-  });
+  const ellapsedTimeRule = (ecsData.signal?.rule?.from != null
+    ? ecsData.signal?.rule?.from[0]
+    : 'now-0s'
+  )
+    .replace('now-', '')
+    .replace('s', '');
+
+  const from = moment(ecsData.timestamp ?? new Date())
+    .subtract(ellapsedTimeRule, 's')
```

Comment:
moment cannot deal with strings such as `now-0s`, that is Elastic Date Math and not moment useable.

You have to parse it out by importing:

```
@elastic/datemath
```

from:
https://github.com/elastic/kibana/tree/master/packages/elastic-datemath

and then using that.

Reply:
If you use that library you shouldn't have to do any of the `.replace` etc... stuff to work around things.

For example, pre-packaged rules do not use seconds, they use math that is of the form `now-6m` so the replacements wouldn't work on some use cases. Using the elastic-datemath I think is the right move here.

Reply:
I did that because I keow the UI is only giving you second, but you are right I forget about the pre-packaged rule